packages/client/hmi-client/src/workflow/ops/model-from-document/tera-model-from-doucment-drilldown.vue

```diff
@@ -0,0 +1,462 @@
+						>
+							<template #header>
+								<h5>{{ equation.name }}</h5>
+							</template>
+							<div class="block-container">
+								<template v-if="instanceOfEquationFromImageBlock(equation.asset)">
+									<label>Extracted Image:</label>
+									<Image
+										id="img"
+										:src="getAssetUrl(equation as AssetBlock<EquationFromImageBlock>)"
+										:alt="''"
+										preview
+									/>
+								</template>
+								<template v-if="!equation.asset.extractionError">
+									<label>Interpreted As:</label>
+									<tera-math-editor :latex-equation="equation.asset.text" :is-editable="false">
+									</tera-math-editor>
+									<InputText v-model="equation.asset.text" />
+								</template>
+								<span v-else>Could not extract LaTeX for image</span>
+							</div>
+						</tera-asset-block>
+					</div>
+				</div>
+				<div v-if="activeStepperIndex === 1">
+					<Textarea v-model="clonedState.text" autoResize disabled style="width: 100%" />
+				</div>
+				<template #footer>
+					<span style="margin-right: auto"
+						><label>Model framework:</label
+						><Dropdown
+							class="w-full md:w-14rem"
+							v-model="clonedState.modelFramework"
+							:options="modelFrameworks"
+							@change="onChangeModelFramework"
+					/></span>
+					<Button
+						label="Run"
+						@click="onRun"
+						:diabled="assetLoading"
+						:loading="loadingModel"
+						outlined
+					></Button>
+				</template>
+			</tera-drilldown-section>
+		</div>
+		<template #preview>
+			<tera-drilldown-preview
+				:options="outputs"
+				v-model:output="selectedOutputId"
+				@update:output="onUpdateOutput"
+				@update:selection="onUpdateSelection"
+				:is-loading="loadingModel"
+				is-selectable
+			>
+				<section v-if="selectedModel">
+					<tera-model-diagram :model="selectedModel" :is-editable="false"></tera-model-diagram>
+					<tera-model-semantic-tables :model="selectedModel" readonly />
+				</section>
+				<tera-operator-placeholder
+					v-else
+					:operation-type="node.operationType"
+					style="height: 100%"
+				/>
+				<template #footer>
+					<Button
+						style="margin-right: auto"
+						label="Save as new model"
+						:disabled="!selectedModel"
+						outlined
+						:loading="savingAsset"
+						@click="isNewModelModalVisible = true"
+					></Button>
+					<Button label="Close" @click="emit('close')"></Button>
+				</template>
+			</tera-drilldown-preview>
+		</template>
+	</tera-drilldown>
+	<tera-modal v-if="isNewModelModalVisible">
+		<template #header>
+			<h4>New Model</h4>
+		</template>
+		<form @submit.prevent>
+			<label for="new-model">Enter a unique name for your model</label>
+			<InputText id="new-model" type="text" v-model="newModelName" placeholder="New model" />
+		</form>
+		<template #footer>
+			<Button @click="saveAsNewModel">Create model</Button>
+			<Button class="p-button-secondary" @click="isNewModelModalVisible = false">Cancel</Button>
+		</template>
+	</tera-modal>
```

Comment:
There is a modal already available for that purpose, you do not have to recreate it.

Reply:
the `tera-model-modal` only creates a new model.  The issue here is that the model has already been made, we are jsut adding it to the project and giving it a new name.  I have added some functionality to the `tera-model-modal` so that we can either create a new model or update a current one with a new name.

Reply:
I'm sorry I thought they were a common modal for a new model like we did for project: https://github.com/DARPA-ASKEM/terarium/blob/main/packages/client/hmi-client/src/components/widgets/tera-common-modal-dialogs.vue

So sorry, this will need to be a new GitHub issue on itself.

Reply:
I fixed it when you mentioned the comment, the `tera-model-modal` does both now 😄 and is now being used here.  If the `tera-model-modal` has a model-id prop it will know to update the name rather then create an empty new model